cmd/acra-server/acra-server.go

```diff
@@ -123,6 +124,8 @@ func main() {
 	tlsDbSNIOld := flag.String("tls_db_sni", "", "Expected Server Name (SNI) from database (deprecated, use \"tls_database_sni\" instead)")
 	tlsDbCert := flag.String("tls_database_cert", "", "Path to client TLS certificate shown to database during TLS handshake (overrides \"tls_cert\")")
 	tlsDbKey := flag.String("tls_database_key", "", "Path to private key of the TLS certificate used to connect to database (see \"tls_database_cert\")")
+	tlsUseClientIDFromCertificate := flag.Bool("tls_client_id_from_cert", false, "Use certificate info as clientID for encryption/decryption when used. Use certificate from connector's connection if acraconnector_tls_transport_enable used or from client's connection if acraconnector_transport_encryption_disable used")
+	tlsIdentifierExtractorType := flag.String("tls_identifier_extractor_type", network.IdentifierExtractorTypeDistinguishedName, fmt.Sprintf("Type of identifier extractor from TLS certificate to use as ClientID (%s|%s)", network.IdentifierExtractorTypeDistinguishedName, network.IdentifierExtractorTypeSerialNumber))
```

Comment:
How about making a list of all extractor types and formatting it here:

```go

fmt.Sprintf("...as ClientID (%s)", strings.Join(IdentifierExtractorTypes, "|"))

var IdentifierExtractorTypes []string = []string{
	IdentifierExtractorTypeDistinguishedName,
	IdentifierExtractorTypeSerialNumber,
}
```

That way it will be a bit more obvious and easy thing to add a new extractor type.

In some intermediate commit this has been located directly in the `network/tls_authentication.go` so it could be kinda okay, but now developers can easily forget to edit `cmd/acra-server/acra-server.go` (and whatever other files there are) when they add a new identifier extractor type.

Reply:
agree, added